subsystems/win32/csrsrv/procsup.c

```diff
@@ -994,8 +994,9 @@ CsrLockProcessByClientId(IN HANDLE Pid,
     *CsrProcess = NULL;
 
     /* Setup the List Pointers */
-    NextEntry = &CsrRootProcess->ListLink;
-    do
+    for (NextEntry = CsrRootProcess->ListLink.Flink;
```

Comment:
How do you know that CsrLockProcessByClientId can never be called with the client id of the root process?
Please don't change code because you're guessing.

Reply:
Actually it does; the "head" of the list CsrRootProcess does point to the data of the root (CSR) process, and it gets called.
If you apply your patch and test ReactOS with it, you should see problems (actually we can make a testbot run to see this).

Reply:
To add extra information, there are routines in the CSR code where the more common while(){} loop (equivalent to the suggested for-loop) is used; this is for code that has to act on all user processes maintained by CSR, except CSR itself.

Reply:
So what you're saying is we should close this PR because it's wrong.

Reply:
I understand my mistake now. I apologize.
Thanks for the explanation(s)! I would suggest to improve/add comments about these different cases...